packages/flutter_tools/lib/src/mdns_discovery.dart

```diff
@@ -171,7 +172,6 @@ Future<Uri> buildObservatoryUri(
   if (!path.endsWith('/')) {
     path += '/';
   }
-  final int localPort = observatoryPort
-      ?? await device.portForwarder.forward(devicePort);
+  final int localPort = await device.portForwarder.forward(devicePort, hostPort: hostVmservicePort);
```

Comment:
In some other places, the name `actualHostPort` is used, which I think might be a little more clear.

Reply:
Done